ArduPlane/tailsitter.cpp

```diff
@@ -267,19 +275,29 @@ bool QuadPlane::tailsitter_transition_vtol_complete(void) const
 {
     if (plane.fly_inverted()) {
         // transition immediately
+        gcs().send_text(MAV_SEVERITY_INFO, "Transition VTOL done, inverted flight");
         return true;
     }
     // for vectored tailsitters at zero pilot throttle
-    if ((plane.quadplane.get_pilot_throttle() < .05f) && plane.quadplane._is_vectored) {
+    if (plane.control_mode->is_vtol_man_throttle() && (plane.quadplane.get_pilot_throttle() < .05f) && plane.quadplane._is_vectored) {
         // if we are not moving (hence on the ground?) or don't know
         // transition immediately to tilt motors up and prevent prop strikes
         if (ahrs.groundspeed() < 1.0f) {
+            gcs().send_text(MAV_SEVERITY_INFO, "Transition VTOL done, zero throttle");
             return true;
         }
     }
-    if (labs(plane.ahrs.pitch_sensor) > tailsitter.transition_angle*100 ||
-        labs(plane.ahrs.roll_sensor) > tailsitter.transition_angle*100 ||
-        AP_HAL::millis() - transition_start_ms > 2000) {
+    const float trans_angle = get_tailsitter_transition_angle_vtol();
+    if (labs(plane.ahrs.pitch_sensor) > trans_angle*100) {
+        gcs().send_text(MAV_SEVERITY_INFO, "Transition VTOL done");
+        return true;
+    }
+    if (labs(plane.ahrs.roll_sensor) >  4500) {
```

Comment:
this looks like a behavior change?

Reply:
Yes, it is. Now the two transition threshold angles are separate there is more chance that one will be low and would instantly be triggered by roll.

Reply:
this should be compared against lim_roll_cd I think....not 45 deg....you could be in a normal turn at 55 deg (most of my planes are setup for 60 deg lim_roll_cd) and you just want a smooth transition rather than a snap....I tested that in my branch, the smooth roll to zero during the transition from 50 deg bank when transition angle was 65...snapping gives lots of wobbles in QLOITER, should be last resort I think...

Reply:
I have changed this to the larger of 45deg and the roll limit + 5